### DISCUSSION TITLE
DFSU to DFS2

### DISCUSSION THREAD

**Discussion by Sparrris** [DHI/mikeio#485] [Q&A]
Hi!

I'm kind of a Python rookie, and I'm working on a script that converts DFSU result files to DFS2 and ASCII which so far works great, but I'm having some problems with the conversion from DFSU to DFS2 using interp_like() taking a lot of time. For example, when running my script this task takes almost 2 minutes  for a particular result file (for just 1 item), whereas it only takes about 15 seconds when doing the same with MIKE Zero (with several items). The code is basically as below, and I'm wondering if I'm doing something wrong or if interp_like() is just a "slow function"?

				grid = ds[0].geometry.get_overset_grid(dx=1, dy=1)
				MaxH_dfs2 = filepath + filename + "_MaxH.dfs2"
				MaxH_grid = ds["Maximum Water Depth"].interp_like(grid)
				MaxH_grid.to_dfs(MaxH_dfs2)

Thanks!

**Comment by ecomodeller** [comment]
The [design principles for MIKE IO](https://dhi.github.io/mikeio/design.html#design-philosophy) have so far been ease of use, and the ability to compose a reproducible and more flexible workflow than using Mike Zero. That said, nobody wants to wait for processing longer than necessary.

The interpolation speed could probably be improved, see this issue as well https://github.com/DHI/mikeio/issues/413

It would also be helpful to understand the context of this conversion, my guess would urban/inland flooding, in order to know which parts to focus on for a potential improvement.
